cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaConnectConfigurationBuilder.java

```diff
@@ -269,20 +278,53 @@ private void printConfigProviders(AbstractConfiguration userConfig) {
      * Adds user provided Kafka Connect configurations.
      *
      * @param configurations   User provided Kafka Connect configurations
-     *
+     * @param injectKafkaJmxReporter         Flag to indicate if metrics are enabled. If they are we inject the JmxReporter into the configuration
+     * @param injectStrimziMetricsReporter   Inject the Strimzi Metrics Reporter into the configuration
      * @return Returns the builder instance
      */
-    public KafkaConnectConfigurationBuilder withUserConfigurations(AbstractConfiguration configurations) {
+    public KafkaConnectConfigurationBuilder withUserConfiguration(AbstractConfiguration configurations,
+                                                                  boolean injectKafkaJmxReporter,
+                                                                  boolean injectStrimziMetricsReporter) {
+        // creating a defensive copy to avoid mutating the input config
+        if (configurations instanceof KafkaConnectConfiguration) {
+            configurations = new KafkaConnectConfiguration((KafkaConnectConfiguration) configurations);
+        } else if (configurations instanceof KafkaMirrorMaker2Configuration) {
+            configurations = new KafkaMirrorMaker2Configuration((KafkaMirrorMaker2Configuration) configurations);
+        } else {
+            configurations = new KafkaConnectConfiguration(reconciliation, new ArrayList<>());
+        }
+
         printConfigProviders(configurations);
 
-        if (configurations != null && !configurations.getConfiguration().isEmpty()) {
-            printSectionHeader("Provided configurations");
+        maybeAddMetricReporters(configurations, injectKafkaJmxReporter, injectStrimziMetricsReporter);
+
+        if (!configurations.getConfiguration().isEmpty()) {
+            printSectionHeader("User Provided configurations with Strimzi Injections");
             writer.println(configurations.getConfiguration());
             writer.println();
         }
         return this;
     }
 
+    // the JMX reporter needs to be explicitly added when metric.reporters is not empty
+    private void maybeAddMetricReporters(AbstractConfiguration userConfig, boolean injectKafkaJmxReporter, boolean injectStrimziMetricsReporter) {
+        if (injectKafkaJmxReporter) {
+            String values = "org.apache.kafka.common.metrics.JmxReporter";
+            // we also need to explicitly configure Kafka clients in this case
+            ModelUtils.createOrAddListConfig(userConfig, "metric.reporters", values);
+            ModelUtils.createOrAddListConfig(userConfig, "admin.metric.reporters", values);
+            ModelUtils.createOrAddListConfig(userConfig, "producer.metric.reporters", values);
+            ModelUtils.createOrAddListConfig(userConfig, "consumer.metric.reporters", values);
+        }
+        if (injectStrimziMetricsReporter) {
+            String values = "io.strimzi.kafka.metrics.KafkaPrometheusMetricsReporter";
```

Comment:
```suggestion
            String values = StrimziMetricsReporterConfig.KAFKA_CLASS;
```
and you can probably inline

Reply:
Fixed and inlined